src/components/pages/profile/[name]/registration/steps/Profile/AddProfileRecordView.tsx

```diff
@@ -0,0 +1,311 @@
+    padding: ${theme.space[5]} 0;
+  `,
+)
+
+const SideBarItem = styled.button(
+  ({ theme }) => css`
+    font-size: ${theme.space['3.5']};
+    line-height: ${theme.space[4]};
+    padding: ${theme.space[2]} ${theme.space[2]};
+  `,
+)
+
+const OptionsContainer = styled.div(
+  ({ theme }) => css`
+    position: relative;
+    flex: 1;
+    overflow: hidden;
+
+    > div {
+      height: 100%;
+      padding-right: ${theme.space[1]};
+      padding-top: ${theme.space[4]};
+      padding-bottom: ${theme.space[4]};
+    }
+  `,
+)
+
+const GroupLabel = styled.div(
+  ({ theme }) => css`
+    padding: ${theme.space[2]} 0;
+    font-weight: ${theme.fontWeights.bold};
+    font-size: ${theme.space['3.5']};
+    line-height: ${theme.space[5]};
+    color: ${theme.colors.textTertiary};
+  `,
+)
+const OptionsGrid = styled.div(
+  ({ theme }) => css`
+    display: grid;
+    grid-template-columns: repeat(auto-fill, minmax(${theme.space['28']}, 1fr));
+    column-gap: ${theme.space[2]};
+    row-gap: ${theme.space[2]};
+  `,
+)
+
+const NoResultsContainer = styled.div(
+  () => css`
+    display: flex;
+    flex-direction: column;
+  `,
+)
+const NoResultsMessage = styled.div(
+  ({ theme }) => css`
+    display: flex;
+    width: 100%;
+    align-items: center;
+    justify-content: center;
+    padding: ${theme.space[4]};
+  `,
+)
+
+const FooterWrapper = styled.div(
+  ({ theme }) => css`
+    border-top: 1px solid ${theme.colors.border};
+    padding: ${theme.space[4]};
+    padding-bottom: 0;
+    margin: 0 -${theme.space['3.5']};
+  `,
+)
+
+type Props = {
+  control: Control<RegistrationForm, any>
+  onAdd?: (records: ProfileRecord[]) => void
+  onClose?: () => void
+}
+
+export const AddProfileRecordView = ({ control, onAdd, onClose }: Props) => {
+  const { t, i18n } = useTranslation('register')
+
+  const currentRecords = useWatch({ control, name: 'records' })
+  const [selectedRecords, setSelectedRecords] = useState<ProfileRecord[]>([])
+
+  const [search, setSearch] = useState('')
+  const [debouncedSearch, _setDebouncedSearch] = useState('')
+  const setDebouncedSearch = useDebouncedCallback(
+    (value: string) => _setDebouncedSearch(value),
+    300,
+  )
+
+  const visibleOptions = useMemo(() => {
+    if (!i18n.isInitialized || !debouncedSearch) return options
+    return options
+      .map((option) => {
+        const items = option.items.filter((item) => {
+          const { key: record } = item
+          if (record.toLowerCase().indexOf(debouncedSearch.toLocaleLowerCase()) !== -1) return true
+          if (['address', 'website'].includes(option.group)) return false
+          const label = t(`steps.profile.options.groups.${option.group}.items.${item}`)
+          return label.toLowerCase().indexOf(debouncedSearch.toLocaleLowerCase()) !== -1
+        })
+        return items.length > 0 ? { ...option, items } : null
+      })
+      .filter((option) => !!option) as typeof options
+  }, [debouncedSearch, t, i18n])
```

Comment:
tbh not sure that this search needs to be debounced, feels awkward in the UI and not necessary since there's no external network reqs